action/governance/createProposal.go

```diff
@@ -124,10 +128,39 @@ func runTx(ctx *action.Context, tx action.RawTx) (bool, action.Response) {
 	//Get Proposal options based on type.
 	options := ctx.ProposalMasterStore.Proposal.GetOptionsByType(createProposal.ProposalType)
 
-	//Calculate Deadlines
-	//Actual voting deadline will be setup in funding Tx
-	fundingDeadline := ctx.Header.Height + options.FundingDeadline
-	votingDeadline := fundingDeadline + options.VotingDeadline
+	//Validate funding goal and pass percentage
```

Comment:
Would you like to move these checks to Validate() and blocks invalid Txs come to mempool?

Reply:
sure, good idea